src/main/scala/com.snowplowanalytics.sauna/responders/urbanairship/UAResponder.scala

```diff
@@ -91,36 +89,52 @@ class UAResponder(urbanairship: urbanAirship)(implicit logger: ActorRef) extends
    * @param appToListMap the map of appKey to list map of values
    * @return Option[l(istToIdentifierMap,appKey)] returns an Option of the tuple if it gets a UAInput object else it return None
    */
-  def getMapEntryForLine(rawLine: String): Option[(Map[String, List[Airship]], String)] = {
+  def getMapEntryForLine(appToListMap: appMap,rawLine: String): Option[appMap] = {
 
     getEntryFromLine(rawLine) match {
-      case Some(uaInput) => Some(getEntry(uaInput))
+      case Some(uaInput) => Some(getEntry(appToListMap,uaInput))
       case _ => None
     }
   }
+  
+  
+  def updateAppMap(appToListMap:appMap,listToAirshipMap:listMap,appKey:String):appMap={appToListMap.filterKeys(_ == appKey).transform((k,v)=>listToAirshipMap)}
```

Comment:
And about these functions.

* You can use signatures I gave you: `updateListMap(listMap: ListMap, uaInput: UAInput): ListMap` and `updateAppMap(appMap: AppMap, uaInput: UAInput): AppMap`. All other arguments seems redundant.
* When you're writing functions/class - please make sure, you're writing it with basic formatting conventions. Use spaces between commas and colons. So this version of function could be written as follows:
```scala
// this much easier to read
def updateAppMap(appToListMap: appMap, listToAirshipMap: listMap, appKey: String): appMap = { 
  appToListMap.filterKeys(_ == appKey).transform((k,v) => listToAirshipMap)
}
```
* But aside from formatting issues, it has few more important: 
  * Signature. As I mentioned above.
  * You don't need to use `filterKeys`. We need here only one key. `filterKeys` assumes we're getting multiple keys.

Also, when you're ready - don't try to refactor other functions alone - we'll do it together. Just these two function: `updateAppMap`, `updateListMap`. Thanks.